tests/test_server.py

```diff
@@ -133,49 +138,113 @@ def test_status_not_ready_agent(rasa_app: SanicTestClient):
 
 
 @pytest.fixture
-def formbot_data():
-    return dict(
-        domain="examples/formbot/domain.yml",
-        config="examples/formbot/config.yml",
-        stories="examples/formbot/data/stories.md",
-        nlu="examples/formbot/data/nlu.md",
-    )
+def shared_statuses() -> DictProxy:
+    return Manager().dict()
 
 
-def test_train_status(rasa_server, rasa_app, formbot_data):
-    with ExitStack() as stack:
-        payload = {
-            key: stack.enter_context(open(path)).read()
-            for key, path in formbot_data.items()
-        }
+@pytest.fixture
+def background_server(
+    shared_statuses: DictProxy, tmpdir: pathlib.Path
+) -> Generator[Process, None, None]:
+    # Create a fake model archive which the mocked train function can return
+    from pathlib import Path
+
+    fake_model = Path(tmpdir) / "fake_model.tar.gz"
+    fake_model.touch()
+    fake_model_path = str(fake_model)
+
+    # Fake training function which blocks until we tell it to stop blocking
+    # If we can send a status request while this is blocking, we can be sure that the
+    # actual training is also not blocking
```

Comment:
Would it be possible, in the future, to add a similar test to `test_train_status_is_not_blocked_by_training`, but using the real `rasa.train`? This new test is very helpful, but I think it would also be interesting to test our choice of threading vs. processing with the real training function.

Reply:
Ella tried this and it makes the test very fragile cause you don't know how long the training is gonna take on travis and all these timings make the test very shaky.